compiler/rustc_codegen_ssa/src/meth.rs

```diff
@@ -114,9 +114,7 @@ pub(crate) fn get_vtable<'tcx, Cx: CodegenMethods<'tcx>>(
 
     let vtable_alloc_id = tcx.vtable_allocation((ty, trait_ref));
     let vtable_allocation = tcx.global_alloc(vtable_alloc_id).unwrap_memory();
-    let vtable_const = cx.const_data_from_alloc(vtable_allocation);
-    let align = cx.data_layout().pointer_align().abi;
-    let vtable = cx.static_addr_of(vtable_const, align, Some("vtable"));
+    let vtable = cx.static_addr_of(vtable_allocation, Some("vtable"));
```

Comment:
Looks like a historical accident that this is separate from `scalar_to_backend`'s own ability to handle vtables (and only one of them is cached?) - AFAICT resulting in different codegen for:
- `let x = (); &x as &dyn Trait`
- `const { &() as &dyn Trait }`